src/server/server.ml

```diff
@@ -250,8 +250,10 @@ let run_exn (requests, replies) : unit Lwt.t =
           Response.create id msg |> reply
         | Error err' ->
           let err = Error.of_list [err; err'] in
-          let msg = Error.to_string_hum err in
-          warning_f ~section "Ignoring junk request: %s" msg)
+          let str = Error.to_string_hum err in
+          warning_f ~section "Ignoring junk request: %s" str |> ignore;
```

Comment:
a small issue here: you shouldn't ignore thread value, either use `ign_warning_f` function or bind the return value of `warning_f` with `reply` thread.